### PR TITLE
VSCode C/C++ include path and config hints

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
Adds Kconfig dotconfig from NuttX to vscode for better hints.
Adds PX4board define header to expose to vscode
Fix include paths for VSCode pointing to newlib instead of NuttX

### CODE REVIEW

**Comment by PetervdPerk-NXP** [comment]
@dagar buildbot commit is wrong, you might want to check on that

File: platforms/common/CMakeLists.txt [hunk 1]
```diff
@@ -63,4 +63,10 @@ endif()
 add_subdirectory(px4_work_queue)
 add_subdirectory(work_queue)
 
+if("${PX4_PLATFORM}" MATCHES "nuttx")
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/c_cpp_properties.json.nuttx.in ${PX4_SOURCE_DIR}/.vscode/c_cpp_properties.json @ONLY)
```
Comment (dagar) [inline platforms/common/CMakeLists.txt:67]: tab

**Comment by dagar** [comment]
> @dagar buildbot commit is wrong, you might want to check on that

Weird, must be coming from the EKF testing change combined with cmake and a missing gitignore?

File: .vscode/c_cpp_properties.json [hunk 2]
```diff
@@ -3,14 +3,18 @@
...
+                "${workspaceFolder}/**",
             ],
             "defines": [],
             "macFrameworkPath": [],
             "configurationProvider": "ms-vscode.cmake-tools",
             "cppStandard": "c++14",
-            "cStandard": "c11"
+            "cStandard": "c11",
+            "forcedInclude": [
+                 "/__w/PX4-Autopilot/PX4-Autopilot/build/px4_sitl_test/px4_boardconfig.h",
```
Comment (dagar) [inline .vscode/c_cpp_properties.json:14]: What's going on here? Why not use the ${workspaceFolder}?

Comment (dagar) [reply .vscode/c_cpp_properties.json]: Ah, that makes more sense.

**Pull request merged**